src/cargo/core/resolver/mod.rs

```diff
@@ -142,9 +142,7 @@ pub fn resolve(
     let mut past_conflicting_activations = conflict_cache::ConflictCache::new();
 
     let resolver_ctx = loop {
-        let resolver_ctx = ResolverContext::new();
```

Comment:
Thinking about this more, I wonder if this was a (failed) attempt to reuse memory. Perhaps `let mut resolver_ctx = ResolverContext::new();` outside the loop and `resolver_ctx.clear()`  inside the loop was the original intent.